docs/resources/cloud_discovery_provider.md

```diff
@@ -0,0 +1,293 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "bloxone_cloud_discovery_provider Resource - terraform-provider-bloxone"
+subcategory: "Cloud Discovery Providers"
+description: |-
+  Manages Cloud Discovery Providers.
+---
+
+# bloxone_cloud_discovery_provider (Resource)
+
+Manages Cloud Discovery Providers.
+
+## Example Usage
+
+```terraform
+resource "bloxone_dns_view" "example" {
+  name = "example_dns_view"
+}
+
+
+resource "bloxone_cloud_discovery_provider" "example_aws" {
+  name               = "example_provider_aws"
+  provider_type      = "Amazon Web Services"
+  account_preference = "single"
+  credential_preference = {
+    access_identifier_type = "role_arn"
+    credential_type        = "dynamic"
+  }
+  source_configs = [
+    {
+      credential_config = {
+        access_identifier = "arn:aws:iam::123456789012:role/role-name"
+      }
+    }
+  ]
+
+  # Other Optional fields
+  destinations = [
+    {
+      config           = {}
+      destination_type = "IPAM/DHCP"
+    },
+    {
+      config = {
+        view_id = bloxone_dns_view.example.id
+      }
+      destination_type = "DNS"
+    }
+  ]
+
+  tags = {
+    site = "Site A"
+  }
+
+}
+
+resource "bloxone_cloud_discovery_provider" "example_azure" {
+  name               = "example_provider_azure"
+  provider_type      = "Microsoft Azure"
+  account_preference = "single"
+  credential_preference = {
+    access_identifier_type = "tenant_id"
+    credential_type        = "dynamic"
+  }
+  source_configs = [
+    {
+      credential_config = {
+        access_identifier = "xyz98765-4321-abcd-efgh-ijklmnopqrst"
+      }
+      restricted_to_accounts = ["12345678-abcd-efgh-ijkl-901234567890"]
+    }
+  ]
+
+  # Other Optional fields
+
+  tags = {
+    site = "Site A"
+  }
+}
+
+resource "bloxone_cloud_discovery_provider" "example_gcp" {
+  name               = "example_provider_gcp"
+  provider_type      = "Google Cloud Platform"
+  account_preference = "single"
+  credential_preference = {
+    access_identifier_type = "project_id"
+    credential_type        = "dynamic"
+  }
+  source_configs = [
+    {
+      credential_config = {
+        access_identifier = "my-bloxone-example-2024"
+      }
+    }
+  ]
+
+  # Other Optional fields
+
+  tags = {
+    site = "Site A"
+  }
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `account_preference` (String) Account preference. For ex.: single, multiple, auto_discover_multiple.
+- `name` (String) Name of the discovery config.
+- `provider_type` (String) Provider type. Ex.: Amazon Web Services, Google Cloud Platform, Microsoft Azure.
+
+### Optional
+
+- `additional_config` (Attributes) Additional configuration. Ex.: '{    "excluded_object_types": [],    "exclusion_account_list": [],    "zone_forwarding": "true" or "false" }'. (see [below for nested schema](#nestedatt--additional_config))
```

Comment:
The "Ex. ...  doesn't look properly formatted in the documentation. You can try to include it in the examples above and remove it from here. There are other fields as well

Reply:
Fixed